components/support/migration/src/main/java/mozilla/components/support/migration/FennecMigrator.kt

```diff
@@ -494,4 +515,23 @@ class FennecMigrator private constructor(
             Result.Failure(e)
         }
     }
+
+    private fun migrateSharedPrefs(): Result<SettingsMigrationResult> {
+        val result = FennecSettingsMigration.migrateSharedPrefs(context)
+
+        if (result is Result.Failure<SettingsMigrationResult>) {
+            val migrationFailureWrapper = result.throwables.first() as SettingsMigrationException
+            return when (migrationFailureWrapper.failure) {
+                is SettingsMigrationResult.Failure.EmptyPreference -> {
+                    logger.error("Detected a corrupt state: ${migrationFailureWrapper.failure}")
+                    crashReporter.submitCaughtException(
```

Comment:
I don't think we should report this exception. What if you're just installing this build (e.g. fresh Fenix install)? That's not a problem at all, but as-is, we'll report an exception.

Reply:
Thanks!.
For migrating Preferences, as of now for only telemetry there is just one possible result: `Success`.
Either because this is a fresh install and there's nothing to migrate, either because we just write a new value to app's shared prefs.
In the future the logic may become more convoluted but for now this is as simple as it gets.

Reply:
Simple is good!